src/components/AuthenticationModal.tsx

```diff
@@ -48,22 +49,28 @@ const errorInstructionSet: Instruction[] = [
   { text: 'Please try again 💪', type: 'secondary' }
 ]
 
-const ConfirmWithAuthModal = ({ onConfirm, onClose, usePin = false }: ConfirmWithAuthModalProps) => {
+const AuthenticationModal = ({ onConfirm, onClose, forcePinUsage = false, ...props }: AuthenticationModalProps) => {
   const insets = useSafeAreaInsets()
 
   const [shownInstructions, setShownInstructions] = useState(firstInstructionSet)
   const [encryptedWallet, setEncryptedWallet] = useState<WalletState>()
-  const [shouldHideModal, setShouldHideModal] = useState(false)
 
   const getWallet = useCallback(async () => {
     try {
-      const storedWallet = await getStoredWallet(usePin)
-      const usesBiometrics = usePin ? false : await loadBiometricsSettings()
+      const storedWallet = await getStoredWallet(forcePinUsage)
+
+      // This should never happen, but if it does, inform the user instead of being stuck
+      if (!storedWallet) {
+        Alert.alert('Missing wallet', 'Could not find wallet to authenticate. Please, restart the app')
+        return
+      }
```

Comment:
This component should only be used when we have a wallet and we want to authenticate. So, in the event that this is used without a wallet (which should be never), we should still catch it and make sure the user doesn't get stuck.